src/chart/radar/radarLayout.js

```diff
@@ -38,8 +39,19 @@ export default function (ecModel) {
 
         data.each(function (idx) {
             // Close polygon
-            points[idx][0] && points[idx].push(points[idx][0].slice());
+
+            var _idx = findFirstNotNaNPoint()
+  
+            // Copy the first actual point to the end of the array
+            _idx >= 0 && points[idx].push(points[idx][_idx].slice());
             data.setItemLayout(idx, points[idx]);
+
+            function findFirstNotNaNPoint(){
+                var _value = zrUtil.find(points[idx], function (point) {
```

Comment:
`value` instead of `_value`

Reply:
thank for your very nice teach! I'm really new for git collaboration. I rebased my commits.
but maybe we need some contributing guidelines? 
I didn't actually thing, if I pass NaN value to render, connect line often disappear when I move mouse or hover point. i dont know what cause the bug yet. I will try to do it

ps: very like your video , i can't wait for next.

Reply:
but should we pass a NaN value to render when we not need a value (such as type "-")?